src/FormControl/index.tsx

```diff
@@ -0,0 +1,262 @@
+                } as const),
+                display: "flex",
+                flexDirection: layoutValue({
+                  vertical: "column",
+                  horizontal: "row",
+                } as const),
+              }),
+              React.isValidElement(containerAs) && containerAs.props.className,
+            ),
+            style: {
+              ...props.style,
+              ...(React.isValidElement(containerAs) && containerAs.props.style),
+            },
+          },
+          <>
+            {(label || description) &&
+              React.cloneElement(
+                createElementFromAs(labelSectionAs),
+                {
+                  className: cx(
+                    css({
+                      flex: layoutValue({
+                        vertical: undefined,
+                        horizontal: 1,
+                      } as const),
+                      marginBottom: layoutValue({
+                        vertical: 8,
+                        horizontal: 0,
+                      } as const),
+                      marginRight: layoutValue({
+                        vertical: undefined,
+                        horizontal: 8,
+                      } as const),
+                    }),
+                    React.isValidElement(labelSectionAs) &&
+                      labelSectionAs.props.className,
+                  ),
+                },
+                <>
+                  {React.isValidElement<any>(label) &&
+                    React.cloneElement(label, {
+                      ...label.props,
+                      id: labelId,
+                      htmlFor: id,
+                    })}
+                  {React.isValidElement<any>(description) &&
+                    React.cloneElement(description, {
+                      ...description.props,
+                      id: descriptionId,
+                    })}
+                </>,
+              )}
+
+            {React.cloneElement(
+              createElementFromAs(contentSectionAs),
+              {
+                className: cx(
+                  css({
+                    flex: layoutValue({
+                      vertical: undefined,
+                      horizontal: 1,
+                    } as const),
+                  }),
+                  React.isValidElement(contentSectionAs) &&
+                    contentSectionAs.props.className,
+                ),
+              },
+              <>
+                <div className={css({ position: "relative" })}>
+                  {startAdornment && (
+                    <div
+                      css={{
+                        position: "absolute",
+                        display: "inline-flex",
+                        left: 12,
+                        top: "50%",
+                        transform: "translateY(-50%)",
+                      }}
+                    >
+                      {startAdornment}
+                    </div>
+                  )}
```

Comment:
(Please do merge at will / don't let this slow down delivery)    

The styles around these wrappers for adornments don't look like they can be easily/safely overridden  

Could FormStartAdornment and FormEndAdornment export styled divs that carried these styles by default?

So instead of directly returning children, they could return these styled div that wraps children

Reply:
Yeah you're right that the adornment layout can't be overridden here. That's kind of unintentional and intentional at the same time. I wanted the components themselves to not have any layout styles in them; just what's needed to render them in isolation and then `FormControl` would handle the layout. I provided escape hatches for some of the layout, but obviously not this part.

I think this will need to be re-worked, especially where we don't even make an attempt to measure the size of the adornments to pad the `Input` and they just overlap the `Select`. 

I want to come back to this when we have a use case for it if that's cool.

Reply:
Yup of course, merge at will  

If you don't mind, I am curious to fill out my mental model so I can better know what kind of feedback is useful to provide -- is the motivation for punting on making this portion of the component open for extension due to a desire to defer coding time? My mental model might be missing some pieces to better weigh the tradeoffs of implementing this now

Reply:
From my perspective of consuming components, since iterating on the API for external dependencies is not quite frictionless, if/when a use case for this arises, I would prefer if the convenient thing to do also happened to be the right thing to do. 

If these adornment components' styles are not open to extension/override, then the convenient thing becomes to wrap the child with negative margins to work around that, which is more convenient from a time-and-effort perspective, but the resulting code would feel less than great

Reply:
@cheapsteak I was thinking in terms of "what's the use case and how should we prepare for that?" Going with a straightforward approach of just applying these css properties in `FormStartAdornment` and `FormEndAdornment` and allowing a `className` override seems fair and frictionless so I added it; thanks for the pushback!